samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -37,49 +36,50 @@
 
   public static final String KEY_NAME = "__key__";
 
-  private final List<Object> value = new ArrayList<>();
-  private final List<Object> relFieldValues = new ArrayList<>();
-  private final List<String> names = new ArrayList<>();
+  private final List<Object> fieldValues = new ArrayList<>();
+  private final List<String> fieldNames = new ArrayList<>();
   private final Object key;
 
   /**
-   * Create the SamzaSqlRelMessage, Each rel message represents a row in the table.
-   * So it can contain a key and a list of fields in the row.
-   * @param key Represents the key in the row, Key is optional, in which case it can be null.
-   * @param names Ordered list of field names in the row.
-   * @param values Ordered list of all the values in the row. Since the samzaSqlRelMessage can represent
-   *               the row in a change capture event stream, It can contain delete messages in which case
-   *               all the fields in the row can be null.
+   * Creates a {@link SamzaSqlRelMessage} from the list of relational fields and values.
+   * If the field list contains KEY, then it extracts the key out of the fields to creates a
+   * RelMessage with key and values otherwise creates a Relmessage without the key.
```

Comment:
Could you add a comment saying that key could be null and fieldValues could be null ? This would be the case for outer joins.